BM-1018: migrating from Bonsai? adoption guide (#712)
boundless-xyz/boundless: documentation/site/pages/developers/tutorials/bonsai.mdx

---
title: Migrating from Bonsai?
description: A practical guide for developers transitioning from Bonsai to Boundless
---

# Migrating from Bonsai?

## What happened to Bonsai?

Bonsai is no longer available for individual developers. Boundless is intended to be a replacement for all your proving needs. If you want to read about the differences between Bonsai and Boundless, please continue on this page. Otherwise, if you have a guest program ready and you are eager to get started, you can follow the [Request a Proof](/developers/tutorials/request) tutorial to create your first proof request.

> Adopting Boundless means embracing decentralization; while the initial setup is more complex than Bonsai's  configuration, you gain access to verifiable compute through permissionless, sustainable proving protocol that aligns with Web3's core principles.

Reviewer: ```suggestion
Adopting Boundless means embracing decentralization; while the initial setup is more complex than Bonsai's  configuration, you gain access to verifiable compute through a highly available, permissionless and self-sustaining proving protocol that aligns with Web3's core principles.
```

## Key Differences

While Bonsai offers simplicity through centralization, Boundless provides:

> - *No vendor lock-in*: Decentralized protocol with multiple provers

Reviewer: Suggestion:

```suggestion
- *Liveness Guarantees*: Decentralized protocol with multiple provers
```

> - *Censorship resistance*: No single point of failure

Reviewer: Suggest
```suggestion
- *High Availability*: No single point of failure
```

- *Long-term sustainability*: Community-driven protocol evolution
- *Transparent pricing*: Market-driven proof costs

> | Aspect | Bonsai | Boundless |

Reviewer: I feel the key difference is that Boundless delivers proofs onchain. I guess thats kind of called out implicitly in Proof Verification, but could be made more explicit by just having Proof Delivery as a row, with onchain/offchian for options

|--------|--------|-----------|
| *Setup* | 2 environment variables (BONSAI_API_URL + BONSAI_API_KEY) | Wallet + Storage Provider  |
| *Proof Format* | STARK/SNARK (Groth16) | Proof aggregated via SetVerifier |
| *Proof Verification* | Via [receipt.verify](https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.Receipt.html#method.verify) or [Verifier Contracts](/developers/smart-contracts/verifier-contracts) | [Verifier Contracts](/developers/smart-contracts/verifier-contracts)|
| *Architecture* | Centralized service | Decentralized protocol |
| *Integration* | Drop-in replacement to guest program | New integration required |

> 

Reviewer 2: Key Benefits
Smart Contract Requestor & Callback features eliminate the need to build off chain infrastructure that can halt the operation of your application.
Proof aggregation amortizes the on chain gas costs for proof verification across the entire Boundless network.
Crytoeconomic guarantees for the time to delivery of your proofs, using slashing and automatic retry logic.
Get your community involved in proving with the industry's only fully open source Multi-GPU Prover stack and a streamlined integration path.
